spec/predicates/vuln.md

```diff
@@ -0,0 +1,136 @@
+
+This document describes a vulnerability attestation type to represent vulnerability reports from the scanners in an "exportable" manner and independently of the format chosen to output the results.
+
+## Prerequisites
+
+The in-toto [attestation] framework and a [Vulnerability scanner tool].
+
+## Use Cases
+
+When sharing the results of a vulnerability scan using an attestation, there is certain metadata that is crucial to trust and reuse this information.
+Information about the scanner used during the scanning is relevant to trust these resuls. The state of the vulnerability database used to search for vulnerabilities defines the accuracy of the results. Other metadata information such as the timestamp when the scan finished could define the reusability of these results.
+
+## Model
+
+This is a predicate type that fits within the larger [Attestation] framework.
+
+## Schema
+
+The schema of this predicate type is documented below.
+
+### Fields
+
+The fields that make up this predicate type are:
+
+The `subject` contains whatever software artifacts are to be associated with this vulnerability report document.
+The `predicate` contains a JSON-encoded data with the following fields:
+
+**scanner**
+
+> There are lots of container image scanners such as Trivy, Grype, Clair, etc.
+> This field describes which scanner is used while performing a container image scan,
+> as well as version information and which Vulnerability DB is used.
+
+**scanner.uri** string (ResourceURI), optional
+
+> > URI indicating the identity of the source of the scanner.
+
+**scanner.version** string (ResourceURI), optional
+
+> The version of the scanner.
+
+**scanner.db.uri** string (ResourceURI), optional
+
+> URI indicating the identity of the source of the Vulnerability DB.
+
+**scanner.db.version** string, optional
+
+> The version of the Vulnerability DB.
+
+**scanner.db.lastUpdate string (Timestamp), required**
+
+> The timestamp of when the vulnerability DB was updated last time.
+
+**scanner.result** object
+
+> This is the most important part of this field because it'll store the scan result as a whole. So, people might want
+> to use this field to take decisions based on them by making use of Policy Engines tooling whether allow or deny these images.
+
+**metadata.scanStartedOn string (Timestamp), required**
+
+> The timestamp of when the scan started.
+
+**metadata.scanFinishedOn string (Timestamp), required**
+
+> The timestamp of when the scan completed.
+
+## Example
+
+```jsonc
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "subject": [
+    {
+      ...
+    }
+  ],
+  // Predicate:
+  "predicateType": "https://in-toto.io/attestation/vulns/attribute-report/v0.1",
```

Comment:
Thanks @hectorj2f this is great. We had almost exactly the same attestation we generated for vulnerability reporting on GUAC: https://github.com/guacsec/guac/blob/main/internal/testing/testdata/exampledata/certify-vuln.json

Once this one finalizes, I would be happy to change over GUAC to use this.